core/sql/sqlcomp/CmpSeabaseDDLrepos.cpp

```diff
@@ -516,6 +585,62 @@ short CmpSeabaseDDL::upgradeReposComplete(ExeCliInterface * cliInterface,
 
         case 1:
           {
+            // If there were views on the old Repository tables, they are
+            // still there, by virtue of the fact that we did "SKIP VIEW CHECK"
+            // on the ALTER TABLE RENAME. Now we will capture their view
+            // definitions (which will contain the old table name, not the
+            // renamed table name) and replay that against the new Repository
+            // table. If the replay fails, we treat that as an unrecoverable
+            // situation and ignore it. Instead, we'll save the view definition
+            // text in the TEXT table with a different TEXT_TYPE and clue the
+            // user in that it is there. They can then try to create the view
+            // at their leisure. Note that they may have to change the view
+            // definition themselves, e.g. if we dropped a column from the
+            // repository table that their view happened to reference.
+            //
```

Comment:
In a future delivery, it might be good to have a feature that returns the view text to the user instead of requiring a read of the TEXT table.    For example, an option to showddl that returns "invalid views". The person fixing up the views may not have SELECT privilege on the system metadata tables.